### PR TITLE
DOCN aquaplanet from file and new A compsets with DOCN modes

### LINKED ISSUE #1581: Add file driven aquaplanet to DOCN
Present DOCN aquaplanet modes are analytic. There is a desire to also have file driven aquaplanet capability as another mode.

### PR DESCRIPTION
**Pull request by mvertens**
Addition of file driven aquaplanet to DOCN

This PR actually does several things:

- adds  aquaplanet capability to DOCN where the input is from a file rather than analytic.

- adds a new directory `$CIMEROOT/src/drivers/mct/cime_config/testdefs/testmods_dirs/drv/default
`
that creates cpl history files on a daily basis and thereby permits answer changes to be compared more easily.

- adds new A compsets to the driver config_compsets.xml that have different DOCN functionality 
ADSOM - DOCN SOM
ADSOMAQP - DOCN aquaplanet SOM
ADAQP3 - DOCN analytic aquaplanet (mode 3)
ADAQFILE - DOCN aquaplanet from file

- removes COPYALL as a possible DOCN_MODE since it is not used

Test suite: scripts_regression tests and
created a modified sandbox off of` 8657a18` with the new compsets and default/ testmods directory to show that the following tests were bfb with `8657a18`
```
ERS.f19_f19.ADAQP3.cheyenne_intel.drv-default.C.8657a18/
ERS.f19_f19.ADSOMAQP.cheyenne_intel.drv-default.C.8657a18/
ERS.f19_g16_rx1.A.cheyenne_intel.drv-default.C.8657a18/
ERS.f19_g16_rx1.ADSOM.cheyenne_intel.drv-default.C.8657a18/
ERS.f19_g16_rx1.AIAF.cheyenne_intel.drv-default.C.8657a18/

```
Test baseline: `8657a18`
Test namelist changes: 
Test status: bit for bit
Fixes #1581 

User interface changes?: None
Code review:

### CODE REVIEW

File: src/drivers/mct/cime_config/config_compsets.xml [hunk 1]
```diff
@@ -46,6 +46,26 @@
   </compset>
 
   <compset>
+    <alias>ADSOM</alias>
+    <lname>2000_DATM%NYF_SLND_DICE%SSMI_DOCN%SOM_DROF%NYF_SGLC_SWAV_TEST</lname>
+  </compset>
+
+  <compset grid="a%1.9x2.5|a%0.9x1.25">
```
Comment (billsacks) [inline src/drivers/mct/cime_config/config_compsets.xml]: Does this grid attribute actually accomplish anything? I tried creating test `SMS.f10_f10.ADSOMAQP.roo2_gnu` and it got through initial case creation just fine, making me think that the grid attribute wasn't doing anything. (I had been expecting the grid attribute to stop me from creating a case with this compset unless I'm using one of the above grids, but that doesn't seem to be the case.)

**Comment by mvertens** [comment]
My understanding was that it should be stopping you from creating a case -
so if it does not - I think this is a bug. Thanks for checking this.

**Comment by billsacks** [comment]
> My understanding was that it should be stopping you from creating a case -
> so if it does not - I think this is a bug. Thanks for checking this.

I checked it because I didn't remember ever seeing `<compset grid=...>` before. But a quick grep through the code shows that it is used in other places. So I'd say this is not a problem with this PR, but rather (maybe) a more general bug. So I approve these changes.

**Comment by mvertens** [comment]
Thanks. Can you please create an issue with this and mark it critical.

**Comment by billsacks** [comment]
@mvertens - opened #1583

File: src/components/data_comps/docn/cime_config/config_component.xml [hunk 2]
```diff
@@ -15,25 +15,25 @@
 
   <entry id="DOCN_MODE">
     <type>char</type>
-    <valid_values>prescribed,sst_aquap1,sst_aquap2,sst_aquap3,sst_aquap4,sst_aquap5,sst_aquap6,sst_aquap7,sst_aquap8,sst_aquap9,sst_aquap10,som,som_aquap,copyall,interannual,null</valid_values>
```
Comment (rljacob) [inline src/components/data_comps/docn/cime_config/config_component.xml]: Did you mean to get rid of "copyall"

File: src/components/data_comps/docn/cime_config/namelist_definition_docn.xml [hunk 3]
```diff
@@ -254,7 +256,7 @@
     <type>char</type>
     <category>streams</category>
     <group>shr_strdata_nml</group>
-    <valid_values>SSTDATA,SST_AQUAP1,SST_AQUAP2,SST_AQUAP3,SST_AQUAP4,SST_AQUAP5,SST_AQUAP6,SST_AQUAP7,SST_AQUAP8,SST_AQUAP9,SST_AQUAP10,SOM,SOM_AQUAP,IAF,NULL,COPYALL</valid_values>
+    <valid_values>SSTDATA,SST_AQUAP1,SST_AQUAP2,SST_AQUAP3,SST_AQUAP4,SST_AQUAP5,SST_AQUAP6,SST_AQUAP7,SST_AQUAP8,SST_AQUAP9,SST_AQUAP10,SST_AQUAPFILE,SOM,SOM_AQUAP,IAF,NULL,COPYALL</valid_values>
```
Comment (rljacob) [inline src/components/data_comps/docn/cime_config/namelist_definition_docn.xml]: COPYALL wasn't removed here but is removed in other parts of this PR

File: src/drivers/mct/cime_config/testdefs/testmods_dirs/drv/default/shell_commands [hunk 4]
```diff
@@ -0,0 +1,2 @@
+./xmlchange HIST_OPTION=ndays
+./xmlchange HIST_N=1
```
Comment (rljacob) [inline src/drivers/mct/cime_config/testdefs/testmods_dirs/drv/default/shell_commands]: Not sure what github is complaining about here.  Missing end-of-file?

Comment (billsacks) [inline src/drivers/mct/cime_config/testdefs/testmods_dirs/drv/default/shell_commands]: I took that symbol to mean "no newline at end of file"

File: src/components/data_comps/docn/cime_config/config_component.xml (hunk 2 above)
Comment (mvertens) [inline src/components/data_comps/docn/cime_config/config_component.xml]: Yes. `copyall `was not being used so having it there was simply confusing.

File: src/components/data_comps/docn/cime_config/namelist_definition_docn.xml (hunk 3 above)
Comment (mvertens) [inline src/components/data_comps/docn/cime_config/namelist_definition_docn.xml]: The datamode in namelist_definition.xml and the DOCN_MODE are not the same thing. The DOCN_MODE maps to a datamode that is used directly by the docn model. For now DOCN_MODE copyall is not used - so I removed it. But the DOCN datamodel value of COPYALL could still be used at some point - so it should stay in the valid values.

File: src/drivers/mct/cime_config/testdefs/testmods_dirs/drv/default/shell_commands (hunk 4 above)
Comment (mvertens) [inline src/drivers/mct/cime_config/testdefs/testmods_dirs/drv/default/shell_commands]: I just added a new line.

File: src/components/data_comps/docn/cime_config/namelist_definition_docn.xml (hunk 3 above)
Comment (rljacob) [inline src/components/data_comps/docn/cime_config/namelist_definition_docn.xml]: Ok.  Just update the PR to mention that COPYALL is being removed from DOCN_MODE.

**Comment by mvertens** [comment]
@billsacks - I have just addressed the comments made by rljacob - so I think this is ready to be merged.

**Pull request merged**